mac_installer/PostInstall.cpp

```diff
@@ -1220,6 +1224,95 @@ Boolean CheckDeleteFile(char *name)
     return false;
 }
 
+
+// If there are other copies of BOINC Manager with different branding
+// on the system, Noitifications may display the icon for the wrong
+// branding, due to the Launch Services database having one of the
+// other copies of BOINC Manager as the first entry. Each user has
+// their own copy of the Launch Services database, so this must be
+// done for each user.
+//
+// This probably will happen only on BOINC development systems where
+// Xcode has generated copies of BOINC Manager.
+static void FixLaunchServicesDataBase(uid_t userID, long brandID) {
+    uid_t saved_uid;
+    char boincPath[MAXPATHLEN];
+    char cmd[MAXPATHLEN+250];
+    long i, n;
+    CFArrayRef appRefs = NULL;
+    OSStatus err;
+
+    if (compareOSVersionTo(10, 8) < 0) {
+        return;  // Notifications before OS 10.8 just bounce our Dock icon
+    }
+
+    saved_uid = geteuid();
+    CFStringRef bundleID = CFSTR("edu.berkeley.boinc");
+
+    if (LSCopyApplicationURLsForBundleIdentifier) { // Weak linked; not available before OS 10.10
+        seteuid(userID);    // Temporarily set effective uid to this user
+        appRefs = LSCopyApplicationURLsForBundleIdentifier(bundleID, NULL);
+        seteuid(saved_uid);     // Set effective uid back to privileged user
+            if (appRefs == NULL) {
+                printf("Call to LSCopyApplicationURLsForBundleIdentifier returned NULL\n");
+                goto registerOurApp;
+            }
+            n = CFArrayGetCount(appRefs);   // Returns all results at once, in database order
+    } else {
+        n = 500;    // Prevent infinite loop
+    }
+
+    for (i=0; i<n; ++i) {     // Prevent infinite loop
+        if (appRefs) {
+            CFURLRef appURL = (CFURLRef)CFArrayGetValueAtIndex(appRefs, i);
+            boincPath[0] = '\0';
+            if (appURL) {
+                CFRetain(appURL);
+                CFStringRef CFPath = CFURLCopyFileSystemPath(appURL, kCFURLPOSIXPathStyle);
+                CFStringGetCString(CFPath, boincPath, sizeof(boincPath), kCFStringEncodingUTF8);
+                if (CFPath) CFRelease(CFPath);
+                CFRelease(appURL);
+                appURL = NULL;
+            }
+        } else {
+            seteuid(userID);    // Temporarily set effective uid to this user
+            // GetPathToAppFromID() returns only first result from database
```

Comment:
Please removed unused commented code

Reply:
That is not unused code. It is a comment explaining that the the loop is needed because the  GetPathToAppFromID() API (called in the next line) returns only the first result from the database, explaining why the loop is needed.

Reply:
Sorry, misunderstood, my bad